src/store/slices/cards/index.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { Position } from '@models/Position';
+
+type CardSliceState = {
+  positions: Record<string, Position>;
+};
+
+const initialState: CardSliceState = {
+  positions: {},
+};
+
+const cardsSlice = createSlice({
+  name: 'cards',
+  initialState,
+  reducers: {
+    initCardPositions: (state, action: PayloadAction<{ cards: string[] }>) => {
```

Comment:
I guess our reducers should be immutable (performance and testability).

What do you think?

Reply:
Actually they are: redux-toolkit and `createSlice` use Immer underneath and thus even with this syntax they are immutable.
I used redux toolkit because it is the official recommendation in the redux website but that being said, we can talk about it, I agree this syntax can be confusing 🙂

Reply:
Didn't know that, nice to know!